tests/e2e/framework/testInfo.go

```diff
@@ -0,0 +1,172 @@
+	"bufio"
+	"cloud.google.com/go/storage"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+)
+
+var (
+	logsBucketPath = flag.String("logs_bucket_path", "", "Cloud Storage Bucket path to use to store logs")
+)
+
+const (
+	TMP_PREFIX    = "istio.e2e."
+	ID_MAX_LENGTH = 36
+)
+
+type TestInfo struct {
+	RunId         string
+	TestId        string
+	Bucket        string
+	LogBucketPath string
+	LogsPath      string
+}
+
+type TestStatus struct {
+	TestId string    `json:"test_id"`
+	Status int       `json:"status"`
+	RunId  string    `json:"run_id"`
+	Date   time.Time `json:"date"`
+}
+
+func NewTestInfo(testId string) (*TestInfo, error) {
+	bucket := ""
+	logsPath := ""
+	f := flag.Lookup("log_dir")
+	tmpDir := f.Value.String()
+	if tmpDir == "" {
+		var err error
+
+		tmpDir, err = ioutil.TempDir(os.TempDir(), TMP_PREFIX)
+		if err != nil {
+			return nil, err
+		}
+
+	}
+	glog.Infof("Using log dir %s", tmpDir)
+	id, err := generateRunId(testId)
+	if err != nil {
+		return nil, err
+	}
+	if *logsBucketPath != "" {
+		r := regexp.MustCompile(`gs://(?P<bucket>[^\/]+)/(?P<path>.+)`)
+		m := r.FindStringSubmatch(*logsBucketPath)
+		if m != nil {
+			bucket = m[1]
+			logsPath = m[2]
+		} else {
+			return nil, errors.New("Cannot parse logBucketPath flag")
+		}
+	}
+	if err != nil {
+		return nil, errors.New("Could not create a Temporary dir")
+	}
+	// Need to setup logging here
+	return &TestInfo{
+		TestId:        testId,
+		RunId:         id,
+		Bucket:        bucket,
+		LogBucketPath: filepath.Join(logsPath, id),
+		LogsPath:      tmpDir,
+	}, nil
+}
+
+func (t TestInfo) Setup() error {
+	return nil
+}
+
+func (t TestInfo) CreateStatusFile(r int) error {
+	glog.Info("Creating status file")
+	ts := TestStatus{
+		Status: r,
+		Date:   time.Now(),
+		TestId: t.TestId,
+		RunId:  t.RunId,
+	}
+	fp := filepath.Join(t.LogsPath, fmt.Sprintf("%s.json", t.TestId))
+	f, err := os.Create(fp)
+	if err != nil {
+		glog.Errorf("Could not create %s", fp)
+		return err
+	}
+	defer f.Close()
+	w := bufio.NewWriter(f)
+	e := json.NewEncoder(w)
+	e.SetIndent("", "  ")
+	err = e.Encode(ts)
+	w.Flush()
+	glog.Infof("Created Status file %s", fp)
+	return err
+}
+
+func (t TestInfo) uploadDir() error {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		glog.Error("Could not set Storage client")
+		return err
+	}
+	bkt := client.Bucket(t.Bucket)
+
+	uploadFile := func(p string) error {
+		// Relative Path
+		rp, err := filepath.Rel(t.LogsPath, p)
+		if err != nil {
+			return err
+		}
+		rp = filepath.Join(t.LogBucketPath, rp)
+		glog.Infof("Uploading %s to gs://%s/%s", p, rp, t.Bucket)
+		o := bkt.Object(rp)
+		w := o.NewWriter(ctx)
+		defer w.Close()
+		b, err := ioutil.ReadFile(p)
+		_, err = w.Write(b)
+		return err
+	}
+
+	walkFn := func(path string, info os.FileInfo, err error) error {
```

Comment:
That s the definition of the walkFn as definied here:
https://golang.org/pkg/path/filepath/#WalkFunc